src/core/Global.ttslua

```diff
@@ -186,6 +186,19 @@ function tryObjectEnterContainer(container, object)
   return true
 end
 
+-- TTS event for objects that enter zones
+-- used to detect the "token discard zones" beneath the hand zones
+function onObjectEnterZone(zone, enteringObj)
+  if zone.getName() ~= "TokenDiscardZone" then return end
+  if tokenChecker.isChaosToken(enteringObj) then return end
+  
+  if enteringObj.type == "Tile" and enteringObj.getMemo() and enteringObj.getLock() == false then
```

Comment:
Not strictly related to this PR, but I wonder if the memo field is the best way to track tokens like this?  What's the rationale to using it instead of a tag and the GMNotes?

Reply:
Historically, the memo field was implemented for tokens to discern them from objects with gmnotes since a lot of code assume the gmnotes to be card metadata

Reply:
Hmm interesting.  I would have thought it shouldn't matter much since tokens should never really have metadata.  Moreover, I don't love the idea of checking a token or card is of a certain type just by the mere _presence_ of gm notes or memos (restricts a lot of the versatility of those fields).

To me a tag is the cleaner way to go.  That being said, not really germane to this PR though.  I'll approve and maybe open an issue to have this conversation